Add new scenario to 50/50 docs (#620)
GSA-TTS/identity-handbook: _articles/manage-50-50-state.md

* Deploy 1: Change the flag value
* Make any changes to routes, jobs, and persistent data in following deploys as described above.

## JavaScript Requests from API Endpoints

> We can hit 50/50 state errors when JavaScript requests to an API endpoint assume that the data will take a certain shape. If an older instance responds to the request, it could cause the request to fail.

Reviewer: Not sure what the appropriate line length is here, but I think you need to split this line into multiple lines. This will be consistent with the rest of the file.

Author: Thanks for the reminder to check line length.  I applied Andrew's suggestions, and I think line length is consistent now.

<details>
<summary>Screenshot with updated text:</summary>

![50-50StateScenarioDocs](https://github.com/user-attachments/assets/57eab36a-946c-4c71-8b59-8a84522b7e60)

</details>

> We can hit 50/50 state errors when JavaScript requests to an API endpoint assume that the data will take a certain shape. If an older instance responds to the request, it could cause the request to fail.

Reviewer 2: It can also go the other direction, where a newer instance responds to the request, and the older JavaScript isn't prepared to handle that new response shape.

Maybe generalizing it a bit could help.

Also, I think the expected result isn't that the request would fail, but that either the behavior of the application is unpredictable, or the code might throw errors if it assumes a property exists that may not.

```suggestion
We can hit 50/50 state errors when JavaScript requests to an API endpoint assume that the data will take a certain shape. A newer or older instance may respond to the request in a way that the JavaScript isn't prepared to handle. This could cause unpredictable behavior, or may cause the code to throw an error if it assumes that a response property will exist.
```

Author: Thanks for the feedback!  Yeah, I agree with generalizing this.  Good point about the expected result, too.

> To avoid such an error, consider modifying the JavaScript request to be less coupled with the shape of the data that the API returns.

Reviewer: same thing here.

> To avoid such an error, consider modifying the JavaScript request to be less coupled with the shape of the data that the API returns.

Reviewer 2: Maybe we can include an example of what it means to be decoupled.

Another option when adding new endpoints is to deploy the endpoint separately from the JavaScript, so that it's guaranteed to be there regardless of which version of the JavaScript is deployed.

```suggestion
To avoid such an error, consider modifying the JavaScript to be less coupled with the shape of the data that the API returns. For example, you can define the response shape to define its properties as potentially `undefined`, which will also allow TypeScript type-checking to identify and potential errors that could be thrown.

When introducing a new endpoint that you intend to request from, you can avoid 50/50 deployment issues by including the API endpoint in a separate, earlier deployment. Then you can safely assume that the endpoint will be available from your JavaScript code.
```

* Deploy 1: Change the flag value
* Make any changes to routes, jobs, and persistent data in following deploys as described above.

## JavaScript Requests from API Endpoints
We can hit 50/50 state errors when JavaScript requests to an API endpoint assume that the data will take a certain shape. A newer or older instance may respond to the request in a way that the JavaScript isn't prepared to handle. This could cause unpredictable behavior, or may cause the code to throw an error if it assumes that a response property will exist.

> To avoid such an error, consider modifying the JavaScript to be less coupled with the shape of the data that the API returns. For example, you can define the response shape to define its properties as potentially `undefined`, which will also allow TypeScript type-checking to identify and potential errors that could be thrown.

Reviewer 2: Small typo in my earlier suggestion

```suggestion
To avoid such an error, consider modifying the JavaScript to be less coupled with the shape of the data that the API returns. For example, you can define the response shape to define its properties as potentially `undefined`, which will also allow TypeScript type-checking to identify any potential errors that could be thrown.
```
